src/VisualStudio/Xaml/Impl/Implementation/XamlTextViewCreationListener.cs

```diff
@@ -161,7 +162,15 @@ private void OnDocumentMonikerChanged(IVsHierarchy hierarchy, string oldMoniker,
                 project.RemoveSourceFile(oldMoniker);
             }
 
-            project.AddSourceFile(newMoniker);
+            var info = _rdt.Value.GetDocumentInfo(docCookie);
```

Comment:
this I guess always called from Ui thread?

Reply:
Since this is called through a RDT event, I'd think that's the case.

Reply:
It is, yes.